### PR TITLE
Change abbreviation characters

### PR DESCRIPTION
**Pull request by mmitche**
While ... is valid in Windows, .NET does not recognize this as a valid path.  As a result, tools like msbuild will fail.  After abbreviation, change ... to --- to avoid issues

Also fix tests on Windows
Windows has \ as the path separator character, and a significantly shorter MAX_PATH.  Set the cached path length long enough that a test that doesn't want the path to be shortened passes.
Also replace / with \ on Windows (and vice versa for non-Windows platforms) when comparing paths.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java [hunk 1]
```diff
@@ -54,8 +63,11 @@ public void doNothingIfThereIsEnoughRoom() throws Exception {
         FreeStyleProject p = f.createProject(FreeStyleProject.class, "and_a_project_in_it");
         p.setAssignedNode(s);
 
+        // Not enough for anything.
+        setMaxPathLength(s, 4096);
```
Comment (olivergondza) [inline src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java:69]: In this test it should be set so it _is_ enough - comment suggests the opposite. Are you suggesting it is not on Windows (and that is why we need this)?

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java [hunk 2]
```diff
@@ -54,8 +63,11 @@ public void doNothingIfThereIsEnoughRoom() throws Exception {
...
+
         FreeStyleBuild b = p.scheduleBuild2(0).get();
-        assertThat(b.getWorkspace().getRemote(), equalTo(s.getRootPath() + "/workspace/" + p.getFullName()));
+        assertThat(b.getWorkspace().getRemote(), equalTo(formatPath(s.getRootPath() + "/workspace/" + p.getFullName())));
```
Comment (olivergondza) [inline src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java]: Why is this needed? If I am not mistaken, this is implemented as `File.createTempFile("hudson", "test", base);` that should be Windows friendly.

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java [hunk 3]
```diff
@@ -41,10 +41,19 @@
...
 
 public class LocatorTest {
 
     @Rule public JenkinsRule j = new JenkinsRule();
+    private String formatPath(String inputPath) {
+        if(Functions.isWindows()) {
+            return inputPath.replace("/", "\\");
+        }
+        else {
+            return inputPath.replace("\\", "/");
```
Comment (olivergondza) [inline src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java]: Backslash is a valid path character on unix.

**Comment by olivergondza** [comment]
> While ... is valid in Windows, .NET does not recognize this as a valid path. As a result, tools like msbuild will fail. After abbreviation, change ... to --- to avoid issues

That is strange. Can you reference some tracked issue/description of the .NET limitation?

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java (hunk 1 above)
Comment (mmitche) [reply src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java:69]: Sorry, comment should be updated.

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java (hunk 3 above)
Comment (mmitche) [reply src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java]: Ahh okay.  In that case we probably just want to replace the windows one.  It is primarily just for consistency, otherwise you end up with lots of workspace paths like "foo\bar/baz/bop" which is harder to read

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java (hunk 1 above)
Comment (mmitche) [reply src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java:69]: Comment needs to be updated

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java (hunk 2 above)
Comment (mmitche) [reply src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java]: The tests did not pass before on Windows because slash directions were getting mixed around.

**Comment by mmitche** [comment]
I work on the .NET team.  I'll try to track down the source of this issue. I'm pretty sure it's fixed in .NET Core but has been a long standing issue in the standard .NET Framework.

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java (hunk 2 above)
Comment (mmitche) [reply src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java]: AHh yeah I remember.  it is plenty windows friendly either way.  However it puts \ in the Windows paths and / in the unix paths.  Which is why formatPath is needed (otherwise the string comparison fails)

**Comment by mmitche** [comment]
The issue with the formatPath are two things:

1) We are comparing the path strings, so using / in the string doesn't work on Windows (it gets normalized)
2) Somewhere in the test framework (or Jenkins), there are still places that / is being produced, so scoping formatPath to places where we are explicitly using / in the tests wasn't enough.

I'm tracking down the original issue of ... in .NET.

**Comment by mmitche** [comment]
Alright, so I tracked it down.  Appears it was an msbuild issue where they were using their own path normalization vs. the official ones (which work).  However, the issue still stands in most versions of msbuild, and given that msbuild is a pretty common build tool invoked by Jenkins, I think the fix still stands.

It was fixed in the .NET Core version of msbuild here: https://github.com/Microsoft/msbuild/pull/632

**Comment by olivergondza** [comment]
@mmitche, I do not mind putting it in, though please make the comment more specific so we know what tool is it, in what version it is fixed.

File: src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java (hunk 2 above)
Comment (olivergondza) [reply src/test/java/org/jenkinsci/plugins/shortwspath/LocatorTest.java]: I have fixed the `/` vs. `\` issue differently on master.

**Comment by mmitche** [comment]
Ahh didn't realize you had fixed the tests.  Reverted that bit, updated the comment.  Thanks!
